src/components/character-count/character-count.js

```diff
@@ -0,0 +1,226 @@
+  } else {
+    length = text.length
+  }
+  return length
+}
+
+// Generate count message and bind it to the input
+// returns reference to the generated element
+CharacterCount.prototype.createCountMessage = function () {
+  var countElement = this.$textarea
+  var elementId = countElement.id
+  // Check for existing info count message
+  var countMessage = document.getElementById(elementId + '-info')
+  // If there is no existing info count message we add one right after the field
+  if (elementId && !countMessage) {
+    countElement.insertAdjacentHTML('afterend', '<span id="' + elementId + '-info" class="govuk-hint govuk-character-count__message" aria-live="polite"></span>')
+    this.describedBy = countElement.getAttribute('aria-describedby')
+    this.describedByInfo = this.describedBy + ' ' + elementId + '-info'
+    countElement.setAttribute('aria-describedby', this.describedByInfo)
+    countMessage = document.getElementById(elementId + '-info')
+  }
+  return countMessage
+}
+
+// Bind input propertychange to the elements and update based on the change
+CharacterCount.prototype.bindChangeEvents = function () {
+  var $textarea = this.$textarea
+  $textarea.addEventListener('keyup', this.updateCountMessage.bind(this))
+
+  // Bind focus/blur events to start/stop polling
+  $textarea.addEventListener('focus', this.handleFocus.bind(this))
+  $textarea.addEventListener('blur', this.handleBlur.bind(this))
+}
+
+// Speech recognition software such as Dragon NaturallySpeaking will modify the
+// fields by directly changing its `value`. These changes don't trigger events
+// in JavaScript, so we need to poll to handle when and if they occur.
+CharacterCount.prototype.checkIfValueChanged = function () {
+  if (!this.oldValue) this.oldValue = ''
+  if (this.$textarea.value !== this.oldValue) {
+    this.oldValue = this.$textarea.value
+    var boundUpdateCountMessage = this.updateCountMessage.bind(this)
+    boundUpdateCountMessage()
+  }
+}
+
+// Update message box
+CharacterCount.prototype.updateCountMessage = function () {
+  var countElement = this.$textarea
+  var options = this.options
+  var countMessage = this.countMessage
+
+  // Determine the remaining number of characters/words
+  var currentLength = this.count(countElement.value, options)
+  var maxLength = this.maxLength
+  var remainingNumber = maxLength - currentLength
+
+  // Set threshold if presented in options
+  var threshold = 0
+  if (options.threshold) {
+    threshold = options.threshold
+  }
+  var thresholdValue = maxLength * threshold / 100
+  if (thresholdValue > currentLength) {
+    countMessage.classList.add('govuk-character-count__message--disabled')
+  } else {
+    countMessage.classList.remove('govuk-character-count__message--disabled')
+  }
+
+  // Update styles
+  if (remainingNumber < 0) {
+    countElement.classList.add('govuk-textarea--error')
+    if (options.validation) {
+      countElement.parentNode.parentNode.classList.add('govuk-form-group--error')
```

Comment:
I tested this locally be setting `validation: true` and the form group error class gets set on the wrong div so that just needs changing. Would it be possible to do this more robustly by looking for the form-group class and adding the class to that?